src/evict/evict_lru.c

```diff
@@ -993,14 +1015,21 @@ __evict_lru_walk(WT_SESSION_IMPL *session)
 			    candidates < entries / 2;
 			    candidates++)
 				if (__evict_read_gen(
-				    &cache->evict_queue[candidates]) > cutoff)
+				    &evict_queue->evict_queue[candidates]) >
+				    cutoff)
 					break;
-			cache->evict_candidates = candidates;
+			evict_queue->evict_candidates = candidates;
 		}
 	}
 
-	cache->evict_current = cache->evict_queue;
-	__wt_spin_unlock(session, &cache->evict_lock);
+	__wt_spin_unlock(session, &evict_queue->evict_lock);
+	/*
+	 * Now we can set the next queue.
+	 */
```

Comment:
We do already have that, in `evict_walk` where the server thread notes whether the existing queue is empty when it starts filling the other one.  That stat is done at the beginning, without holding any locks.  I will move that code to here.  It will only add a few lines to the lock code (which is still much, much less than the code in `evict_get_ref`).